src/sdl2/render.rs

```diff
@@ -125,40 +178,233 @@ impl RendererInfo {
+        }
+    }
+}
+
+pub type WindowCanvas = Canvas<Window>;
+
+/// Methods for the `WindowCanvas`.
+impl Canvas<Window> {
+    /// Gets a reference to the associated window of the Canvas
+    #[inline]
+    pub fn window(&self) -> &WindowRef {
+        &self.target
+    }
+
+    /// Gets a mutable reference to the associated window of the Canvas
+    #[inline]
+    pub fn window_mut(&mut self) -> &mut WindowRef {
+        &mut self.target
+    }
+
+    /// Gets the associated window of the Canvas and destroys the Canvas
+    #[inline]
+    pub fn into_window(self) -> Window {
+        self.target
+    }
+
+    /// Sets the render target to the provided texture
+    /// Returns a handle for rendering methods to the target
+    /// Returns `TargetRenderError::NotSupported`
+    /// if the renderer does not support the use of render targets,
+    /// Returns `TargetRenderError::SdlError` if SDL2 returned with an error code.
+    /// The texture must be created with the texture access: `sdl2::render::TextureAccess::Target`.
+    pub fn with_target<'r, 't, 'a>(&'r mut self,
+                                   texture: &'t mut Texture<'a>)
+                                   -> Result<TextureCanvas<'r, 't>, TargetRenderError> {
+        self.internal_with_target(texture)
+    }
+
+    /// Returns a `TextureCreator` that can create Textures to be drawn on this `Canvas`
+    ///
+    /// This `TextureCreator` will share a reference to the renderer and target context.
+    ///
+    /// The target (i.e., `Window`) will not be destroyed and the SDL_Renderer will not be
+    /// destroyed if the `TextureCreator` is still in scope.
+    pub fn texture_creator(&self) -> TextureCreator<WindowContext> {
+        TextureCreator {
+            context: self.context.clone(),
+            _target_context: self.target.context(),
+        }
+    }
+}
+
+impl<T> Canvas<T> {
+    /// Determine whether a window supports the use of render targets.
+    pub fn render_target_supported(&self) -> bool {
+        unsafe { ll::SDL_RenderTargetSupported(self.context.raw) == 1 }
+    }
+
+    fn internal_with_target<'r, 't, 'a>(&'r mut self,
+                                        texture: &'t mut Texture<'a>)
+                                        -> Result<TextureCanvas<'r, 't>, TargetRenderError> {
+        if self.render_target_supported() {
+            unsafe { self.set_raw_target(texture.raw) }
+                .map_err(|e| TargetRenderError::SdlError(e))?;
+            Ok(TextureCanvas {
+                   context: self.context.clone(),
+                   target: TextureTarget {
+                       raw_renderer: &self.context.raw,
+                       _texture_marker: PhantomData,
+                   },
+               })
+        } else {
+            Err(TargetRenderError::NotSupported)
+        }
+    }
+}
+
+/// Creates Textures that cannot outlive the creator
+/// Holds its own reference to the target and the renderer context.
```

Comment:
Add a newline between these two.

Add "TextureCreator doesn't hold a lifetime to its Canvas by design choice. If a Canvas is dropped before its TextureCreator, it is still safe to use, but it is however useless (since Texture can only used via Canvas and the linked Canvas has been dropped)." somewhere after or to replace the reference mention.